src/main/java/org/apache/commons/io/FilenameUtils.java

```diff
@@ -1385,11 +1386,11 @@ public static boolean wildcardMatch(final String fileName, final String wildcard
         boolean anyChars = false;
         int textIdx = 0;
         int wcsIdx = 0;
-        final Stack<int[]> backtrack = new Stack<>();
+        final Deque<int[]> backtrack = new ArrayDeque<>(wcs.length);
```

Comment:
Looks reasonable to me. `Stack` is synchronized I think, while `Deque` is not. But `backtrack` is used in this method without any locks, so `Stack` doesn't appear to be used specifically for that.